src/core/friendly_errors/param_validator.js

```diff
@@ -2,15 +2,57 @@
  * @for p5
  * @requires core
  */
-// import p5 from '../main';
-import dataDoc from '../../../docs/parameterData.json';
+import p5 from '../main.js';
+import * as constants from '../constants.js';
 import { z } from 'zod';
 import { fromError } from 'zod-validation-error';
+import dataDoc from '../../../docs/parameterData.json' assert { type: 'json' };
 
 // Cache for Zod schemas
 let schemaRegistry = new Map();
 const arrDoc = JSON.parse(JSON.stringify(dataDoc));
 
+// Mapping names of p5 types to their constructor functions.
+// p5Constructors:
+//   - Color: f()
+//   - Graphics: f()
+//   - Vector: f()
+// and so on.
+const p5Constructors = {};
+// For speedup over many runs. `funcSpecificConstructors[func]` only has the
+// constructors for types which were seen earlier as args of `func`.
+const funcSpecificConstructors = {};
+
+for (let [key, value] of Object.entries(p5)) {
+  p5Constructors[key] = value;
+}
+
+// window.addEventListener('load', () => {
+//   // Make a list of all p5 classes to be used for argument validation
+//   // This must be done only when everything has loaded otherwise we get
+//   // an empty array.
+//   for (let key of Object.keys(p5)) {
+//     // Get a list of all constructors in p5. They are functions whose names
+//     // start with a capital letter.
+//     if (typeof p5[key] === 'function' && key[0] !== key[0].toLowerCase()) {
+//       p5Constructors[key] = p5[key];
+//     }
+//   }
+// });
+
+// `constantsMap` maps constants to their values, e.g.
+// {
+//   ADD: 'lighter',
+//   ALT: 18,
+//   ARROW: 'default',
+//   AUTO: 'auto',
+//   ...
+// }
+const constantsMap = {};
+for (const [key, value] of Object.entries(constants)) {
```

Comment:
Hmm...that's a good question. It might be that module need to specifically register things as constants but I want to avoid that where possible to simplify things.

Reply:
Is it possible to solve rely on the documentation parsed data to get this list? We can possibly add extra tags or into into the JSDoc comments if needed to mark something as a constant.

Reply:
It might not be possibly to get it from the parameter data because the types only include the names of the constants, and we won't know from that what value they correspond to (like what specific symbol to check against)

Reply:
If we switch back to using strings, I think it could be doable though?

Reply:
We can possibly switch back to strings, the symbol stuff I'm not 100% sold on yet as I'm also not sure how or if it should work across multiple instances.